packages/destination-actions/src/destinations/heap/trackEvent/index.ts

```diff
@@ -75,10 +96,16 @@ const action: ActionDefinition<Settings, Payload> = {
 
     const event: HeapEvent = {
       app_id: settings.appId,
-      identity: payload.identity,
       event: payload.event,
       properties: eventProperties,
-      idempotency_key: '1' // :TODO:
+      idempotency_key: payload.message_id
+    }
+    if (payload.anonymous_id && !payload.identity) {
```

Comment:
Should we have a check before this that exits if there is no anonymous_id or identity?